pkg/hs/urbit-king/lib/Urbit/Vere/Dawn.hs

```diff
@@ -168,122 +153,114 @@ dawnPostRequests endpoint responseBuilder requests = do
-                                    , "id"      .= rid
+                                    , "id"      .= (show rid)
                                     ]
 
 -- Azimuth JSON Requests -------------------------------------------------------
 
--- Not a full implementation of the Ethereum ABI, but just the ability to call
--- a method by encoded id (like 0x63fa9a87 for `points(uint32)`), and a single
--- UIntN 32 parameter.
-encodeCall :: Text -> Int -> Text
-encodeCall method idx = method <> leadingZeroes <> renderedNumber
-  where
-    renderedNumber = pack $ showHex idx ""
-    leadingZeroes = replicate (64 - length renderedNumber) '0'
-
-data BlockRequest = BlockRequest
-  deriving (Show, Eq)
-
-instance RequestMethod BlockRequest where
-  getRequestMethod BlockRequest = "eth_blockNumber"
-
-instance ToJSON BlockRequest where
-  toJSON BlockRequest = Array $ fromList []
-
--- No need to parse, it's already in the format we'll pass as an argument to
--- eth calls which take a block number.
-parseBlockRequest :: BlockRequest -> Text -> TextBlockNum
-parseBlockRequest _ txt = txt
-
-type TextBlockNum = Text
-
-data PointRequest = PointRequest
-  { grqHexBlockNum :: TextBlockNum
-  , grqPointId :: Int
-  } deriving (Show, Eq)
+data PointResponse = PointResponse
+  --NOTE  also contains dominion and ownership, but not actually used here
+  { prNetwork :: PointNetwork
+  } deriving (Show, Eq, Generic)
+
+instance FromJSON PointResponse where
+  parseJSON (Object o) = do
+    prNetwork <- o .: "network"
+    pure PointResponse{..}
+  parseJSON _ = do error "failed to parse PointResponse json"
+
+data PointNetwork = PointNetwork
+  { pnKeys :: PointKeys
+  , pnSponsor :: PointSponsor
+  , pnRift :: ContNum
+  } deriving (Show, Eq, Generic)
+
+instance FromJSON PointNetwork where
+  parseJSON (Object o) = do
+    pnKeys <- o .: "keys"
+    pnSponsor <- o .: "sponsor"
+    pnRift <- o .: "rift"
+    pure PointNetwork{..}
+  parseJSON _ = do error "failed to parse PointNetwork json"
+
+data PointKeys = PointKeys
+  { pkLife :: Life
+  , pkSuite :: Int
+  , pkAuth :: Text  --NOTE  0xhex string
+  , pkCrypt :: Text  --NOTE  0xhex string
+  } deriving (Show, Eq, Generic)
+
+instance FromJSON PointKeys where
+  parseJSON (Object o) = do
+    pkLife <- o .: "life"
+    pkSuite <- o .: "suite"
+    pkAuth <- o .: "auth"
+    pkCrypt <- o .: "crypt"
+    pure PointKeys{..}
+  parseJSON _ = do error "failed to parse PointKeys json"
+
+data PointSponsor = PointSponsor
+  { psHas :: Bool
+  , psWho :: Text
+  } deriving (Show, Eq, Generic)
+
+instance FromJSON PointSponsor where
+  parseJSON (Object o) = do
+    psHas <- o .: "has"
+    psWho <- o .: "who"
+    pure PointSponsor{..}
+  parseJSON _ = do error "failed to parse PointSponsor json"
+
+data PointRequest = PointRequest Ship
 
 instance RequestMethod PointRequest where
-  getRequestMethod PointRequest{..} = "eth_call"
+  getRequestMethod _ = "getPoint"
```

Comment:
One advantage of having this be `PointRequest{}` rather than _, as in the beforeside of this diff, is that if you add an extra constructor to the type, you get a warning (incomplete pattern match) saying, in effect, that you need to think about what the request method should be for the new constructor. If you use `_`, it'll silently reuse "getPoint" which may or may not be what you intend; in any event you should be forced to think about this in the hypothetical refactor.